docs/build.md

```diff
@@ -227,6 +229,55 @@ spec:
       name: icr-knbuild
 ```
 
+### Runtime-Image
+
+Runtime-image is a new image composed with build-strategy outcome. On which you can compose a multi-stage image build, copying parts out the original image into a new one. This feature allows replacing the base-image of any container-image, creating leaner images, and other use-cases.
+
+The following examples illustrates how to the `runtime`:
+
+```yml
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: nodejs-ex-runtime
+spec:
+  strategy:
+    name: buildpacks-v3
+    kind: ClusterBuildStrategy
+  source:
+    url: https://github.com/sclorg/nodejs-ex.git
+  output:
+    image: image-registry.openshift-image-registry.svc:5000/build-examples/nodejs-ex
+  runtime:
+    base:
+      image: docker.io/node:latest
+    workDir: /home/node/app
+    run:
+      - echo "Before copying data..."
+    user:
+      name: node
+      group: "1000"
+    paths:
+      - /workspace/source:/home/node/app
+    entrypoint:
+      - npm
+      - start
+```
+
+This build will produce a Node.js based application where a single directory is imported from the image built by buildpacks strategy. The data copied is using the `.spec.runtime.user` directive, and the image also runs based on it.
+
+Please consider the description of the attributes under `.spec.runtime`:
+
+- `.base`: specifies the runtime base-image to be used, using Image as type
+- `.workDir`: path to WORKDIR in runtime-image
+- `.env`: runtime-image additional environment variables, key-value
+- `.labels`: runtime-image additional labels, key-value
+- `.run`: arbitrary commands to be executed as `RUN` blocks, before `COPY`
+- `.user.name`: username employed on `USER` directive, and also to change ownership of files copied to the runtime-image
+- `.user.group`: group name (or GID), employed to change ownership and on `USER` directive
+- `.paths`: list of files or directory paths to be copied to runtime-image, those can be defined as `<source>:<destination>` split by colon (`:`)
+- `.entrypoint`: entrypoint command, specified as a list
+
```

Comment:
Somewhere here should be the big note as discussed in today's meeting:

**Note**: specifying the runtime section will cause a build run to push to your defined `spec.output.image` destination twice. The first image pushed is the image produced by the steps defined in the build strategy. The second image push is the output of the runtime image build. Be aware of this behavior in case you have automated operations (for example a deployment of that image to a cluster) that start when an image gets pushed.

Reply:
I've added this warning note below, please consider.

Reply:
OK.